### PR TITLE
Only passing in error_code on error

### PR DESCRIPTION
**Pull request by RLEE0731**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @RLEE0731 has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m [hunk 1]
```diff
@@ -361,8 +363,10 @@ - (void)writeToKeychain
         }
         result = SecItemAdd((CFDictionaryRef)query, NULL);
         BOXAssert( result == noErr, @"Couldn't add the Keychain Item with error %d.", (int)result);
-        NSDictionary *userInfo = @{@"completion_status" : (result == noErr) ? @"succeeded" : @"failed",
-                                   @"error_code" : @(result)};
+        NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:@{@"completion_status" : (result == noErr) ? @"succeeded" : @"failed"}];
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m]: we can move this to outside the if block.

File: BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m [hunk 2]
```diff
@@ -299,14 +299,22 @@ - (void)overwriteKeychainItem
     if (result != noErr) {
         BOXLog(@"Keychain overwrite failed. Failed to add keychain item with error %d", (int)result);
     }
+    NSMutableDictionary *userInfo = @{@"completion_status" : (result == noErr) ? @"succeeded" : @"failed",
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m]: could we do the same after deleting keychain above result = SecItemDelete...?

File: BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m [hunk 3]
```diff
@@ -299,14 +299,22 @@ - (void)overwriteKeychainItem
...
+                                      @"message" : @"Overwriting all items in keychain"}.mutableCopy;
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m]: "Overwriting items in keychain by adding"

File: BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m (hunk 2 above)
Comment (thuyBox) [reply BoxContentSDK/BoxContentSDK/External/KeychainItemWrapper/BOXKeychainItemWrapper.m]: "Overwriting items in keychain by first deleting"

**Review by thuyBox — APPROVED** [review]

**Pull request merged**